api/starknet_ws_api.json

```
@@ -216,14 +308,14 @@
         },
         {
           "name": "result",
-          "description": "Either a tranasaction hash or full transaction details, based on subscription",
+          "description": "A transaction and its current finality status",
           "schema": {
-            "oneOf": [
+            "allOf": [
               {
-                "$ref": "./api/starknet_api_openrpc.json#/components/schemas/TXN_HASH"
+                "$ref": "./api/starknet_api_openrpc.json#/components/schemas/TXN_WITH_HASH"
               },
               {
-                "$ref": "./api/starknet_api_openrpc.json#/components/schemas/TXN_WITH_HASH"
+                "$ref": "#/components/schemas/TXN_STATUS_WITHOUT_L1"
```

This does not specify the property name that should contain the transaction status.

We should probably use something like this instead:

```suggestion
                "type": "object",
                "properties": {
                  "finality_status": {
                    "description": "Finality status of the transaction",
                    "$ref": "#/components/schemas/TXN_STATUS_WITHOUT_L1"
                  }
                },
                "required": [
                  "finality_status"
                ]
```